FHIRClientDevelopmentGuide.md

```diff
@@ -0,0 +1,100 @@
+
+The client must support polling the FHIR API to regularly check for FHIR message responses from NCHS. It is recommended to poll the FHIR API at least once every 5 minutes.
+
+### Configurable Parameters
+- Submission interval
+- Polling interval
+
+## Message Resends
+The client must support resends if a message is not acknowledged within the configured time frame. The resubmission message header must be identical to the initial submission message header. The header date timestamp must be identical to the initial message to preserve the order the submission message and any update messages should be processed by NCHS. It is recommended to save the json bundle to preserve the exact message in case there is a need to resubmit. The current recommended resend window is 5 hours.
+
+The maximum number of retries should be 3. After 3 attempts and no acknowledgements, a jurisdictional contact at NCHS should be notified via email or some other notification system. Jurisdictions should identify someone on their end of the system who is responsible for monitoring failed messages who will receive and respond to these notifications.
+
+The current recommendation is to implement an “exponential” back off when resending messages. 
+1.	5 hours
+2.	10 hours
+3.	20 hours
+
+### Configurable Parameters
+- Resend interval
+- Maximum retries
+
+## Response Message Handling
+The client must handle the following message response types
+- Acknowledgements (ACK)
+- Extraction Errors (ERR)
+- Cause of Death Coding Responses (TRX)
+- Demographic Coding Responses (MRE)
+- Status Response (STM)
+
+The client must acknowledge the following message response types
+- Cause of Death Coding Responses (TRX)
+- Demographic Coding Responses (MRE)
+
+The client should support acknowledging response messages in a timely manner to avoid unnecessary duplicate messages from NCHS. Acknowledgements should be sent as soon as they become available and no more than 3 hours after.
+
+### Configurable Parameters
+- Submission interval
+
+## Message Traceabality
+The client must maintain a persistent store of Death Records submitted to NCHS. It is recommended to make the message ID of past records easily accessible for traceability.
+
+The client must support tracing message responses to their initial submission so they can track the status of that message. Use the message ID of the submitted message and the reference message ID in message responses to implement traceability. 
+
+The client should provide an easy method to see the status of submitted messages and identify records that resulted in an Extraction Error and require modification.
+
+## NCHS Backup Endpoint
+The client must be configured to submit messages to STEVE by default and to NCHS as a backup when STEVE is unavailable. Jurisdictions should only submit messages to the backup endpoint when NCHS directs them to switch from STEVE to the back up endpoint.
+
+### Configurable Parameters
+- STEVE endpoint
+- NCHS backup endpoint
```

Comment:
Aha, this covers most of an earlier comment... but also maybe allow configuration of which endpoint to use?